data/data/aws/bootstrap/main.tf

```diff
@@ -11,6 +11,24 @@ locals {
   volume_type      = "gp3"
   volume_size      = 30
   volume_iops      = local.volume_type == "io1" ? 100 : 0
+
+  // s3 object supports only 10 tags. The first 9 tags from 
+  // the list of aws_extra_tags are used for s3 object
+  new_tag_len = 9
+
+  keys  = flatten([
+      for key, value in local.tags: "${key}"
+  ])
+  values = flatten([
+      for key, value in local.tags: "${value}"
+  ])
+
+  slicelen = length(local.keys) < local.new_tag_len ? length(local.keys) : local.new_tag_len
+
+  slicedkeys = slice(local.keys, 0, local.slicelen)
+  slicedvalues = slice(local.values, 0, local.slicelen)
+
+  s3_object_tags = zipmap(local.slicedkeys, local.slicedvalues)
```

Comment:
I think we can simplify how to get the first X items from the map. Using `terraform console` I tested out: `{for k in slice(keys(var.longMap), 0, 8): k => var.longMap[k]}`:

```
> var.longMap
tomap({
  "a" = 0
  "b" = 1
  "c" = 2
  "d" = 3
  "e" = 4
  "f" = 5
  "g" = 6
  "h" = 7
  "i" = 8
  "j" = 9
})
> {for k in slice(keys(var.longMap), 0, 8): k => var.longMap[k]}
{
  "a" = 0
  "b" = 1
  "c" = 2
  "d" = 3
  "e" = 4
  "f" = 5
  "g" = 6
  "h" = 7
}
```
So I think we could combine that with a ternary to get the shortened user tags, something like:

```
s3_object_user_tags = length(var.aws_extra_tags) <= 8 ? var.aws_extra_tags : {for k in slice(keys(var.aws_extra_tags), 0, 8): k => var.aws_extra_tags[k]}
```
(I did not test this last command)

Then we just need to ultimately make sure to add in the name tag and the owned tag.